dev/bots/analyze_snippet_code.dart

```diff
@@ -2,10 +2,61 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// See ../snippets/README.md for documentation.
-
 // To run this, from the root of the Flutter repository:
-//   bin/cache/dart-sdk/bin/dart dev/bots/analyze_snippet_code.dart
+//   bin/cache/dart-sdk/bin/dart --enable-asserts dev/bots/analyze_snippet_code.dart
+
+// In general, please prefer using full inline examples in API docs.
+//
+// For documentation on creating sample code, see ../../examples/api/README.md
+// See also our style guide's discussion on documentation and sample code:
+// https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#documentation-dartdocs-javadocs-etc
+//
+// This tool is used to analyze smaller snippets of code in the API docs.
+// Such snippets are wrapped in ```dart ... ``` blocks, which may themselves
+// be wrapped in {@tool snippet} ... {@endtool} blocks to set them apart
+// in the rendered output.
+//
+// Such snippets:
+//
+//  * If they start with `import` are treated as full application samples; avoid
+//    doing this in general, it's better to use samples as described above. (One
+//    exception might be in dart:ui where the sample code would end up in a
+//    different repository which would be awkward.)
+//
+//  * If they start with a comment that says `// continuing from previous example...`,
```

Comment:
I wonder if we should give these control comments a common prefix. It would indicate to the reader that these comments are actually load-bearing. Also, if I want to author a new example and look at an existing one it would give me something to search for to understand what other control comments are available.

On the other hand, it my hinder the reading flow of these examples...

Just a thought, not feeling super-strongly about this either way.

Reply:
yeah i feel super constrained here because the most important result is that the examples flow well for the developers.
on the other hand, these load-bearing comments are so magical, ugh.

I've added a message to the snippet analyzer's output that just points right at the docs, FWIW. Dunno how much it'll help.